TaylorFramework/Modules/scissors/ExtendedComponent.swift

```diff
@@ -56,51 +56,59 @@ final class ExtendedComponent {
         self.components = []
     }
     
-    func appendComponents(components: [ExtendedComponent], array: [SourceKitRepresentable]) -> [ExtendedComponent] {
+    func appendComponents(array: [SourceKitRepresentable], components: NSMutableArray = []) -> [ExtendedComponent] {
```

Comment:
If you were to write this function once again, with less mutating states and maybe composed from shorter functions, what would that look like?